src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -29,34 +29,33 @@ public static unsafe void Fill<T>(ref T refData, nuint numElements, T value)
             {
                 // We have enough data for at least one vectorized write.
 
-                T tmp = value; // Avoid taking address of the "value" argument. It would regress performance of the loops below.
                 Vector<byte> vector;
 
                 if (sizeof(T) == 1)
                 {
-                    vector = new Vector<byte>(Unsafe.As<T, byte>(ref tmp));
+                    vector = new Vector<byte>(Unsafe.BitCast<T, byte>(value));
                 }
                 else if (sizeof(T) == 2)
                 {
-                    vector = (Vector<byte>)(new Vector<ushort>(Unsafe.As<T, ushort>(ref tmp)));
+                    vector = (Vector<byte>)(new Vector<ushort>(Unsafe.BitCast<T, ushort>(value)));
                 }
                 else if (sizeof(T) == 4)
                 {
                     // special-case float since it's already passed in a SIMD reg
                     vector = (typeof(T) == typeof(float))
-                        ? (Vector<byte>)(new Vector<float>((float)(object)tmp!))
-                        : (Vector<byte>)(new Vector<uint>(Unsafe.As<T, uint>(ref tmp)));
+                        ? (Vector<byte>)(new Vector<float>(Unsafe.BitCast<T, float>(value)))
+                        : (Vector<byte>)(new Vector<uint>(Unsafe.BitCast<T, uint>(value)));
                 }
                 else if (sizeof(T) == 8)
                 {
                     // special-case double since it's already passed in a SIMD reg
                     vector = (typeof(T) == typeof(double))
-                        ? (Vector<byte>)(new Vector<double>((double)(object)tmp!))
-                        : (Vector<byte>)(new Vector<ulong>(Unsafe.As<T, ulong>(ref tmp)));
+                        ? (Vector<byte>)(new Vector<double>(Unsafe.BitCast<T, double>(value)))
+                        : (Vector<byte>)(new Vector<ulong>(Unsafe.BitCast<T, ulong>(value)));
```

Comment:
Is this method reachable for e.g. `int?`?
The conditions above look like they'd let it use the vectorized path, but `BitCast` doesn't work with nullable.

Reply:
this whole thing could really be simplified to just `return Vector.Create<T>(value).As<T, byte>();`

Reply:
All of these throw a `NotSupportedException` for `int?`, we have to use `Unsafe.As`.

[sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgDkBXfGKASzFwBuGvSasAShwB2GXtxYBhCPgAOvADY8AyjwBu/GEJGMtAC2xQVAGWzAWkmXJjDqIgMxNSDBQwDeNBkCGAG0AKV4MAHEYKR5+AAoMAE8VGAgAM3jeGQB+AEo8gF0AoOIPADUYMAxoAB5gJIwYAD4GAFlyWoAVZviuhl1sdQ4YPL8SoNKAdgZYgHcGSuq6hqbegFUpXGx0mBYAQVxutAZVlvjYdIGhkYKXSYBfCYZnsIjo2L4wRJS0zOyMPkis8yosqjUoPVGi12qRur1+oNhqNxtRJpNiDMlhDFLBsE14kjbgcjl0Tmdenl7kEnmigq9wlEYnFvslUhksrkCsU6YFQdiVtDWm03PC+tdkWN/Lz0ZjZjAFgLIRT4pttrsWAAhCIKPAYY6nIWEm6jKnPWkPIA===)